solidity/test/vault/TBTCVault.OptimisticMinting.test.ts

```diff
@@ -0,0 +1,1867 @@
+          // The sum of sweep tx inputs is 20000 satoshi. The output value is
+          // 18500 so the transaction fee is 1500. There is only one deposit so
+          // it incurs the entire transaction fee.
+          // Treasury fee is cut when optimistically minting TBTC but given the
+          // Bitcoin transaction fee is unknown at the moment of optimistic
+          // minting, we can not deduct it when optimistically minting TBTC so
+          // the optimistic minting debt stay equal to the transaction fee.
+          expect(
+            await tbtcVault.optimisticMintingDebt(depositRevealInfo.depositor)
+          ).to.equal(1500)
+        })
+
+        it("should emit an event", async () => {
+          await expect(tx)
+            .to.emit(tbtcVault, "OptimisticMintingDebtRepaid")
+            .withArgs(depositRevealInfo.depositor, 1500)
+        })
+      }
+    )
+
+    context("when multiple deposits gets swept", () => {
+      interface Fixture {
+        mockBank: FakeContract<Bank>
+        mockBridge: FakeContract<Bridge>
+        tbtc: TBTC
+        tbtcVault: TBTCVault
+      }
+
+      const depositor = "0xb2Ea9bb14A901fD71A7cf6e7b5bdC62aA2b9F012"
+
+      // Setting up two real testnet deposits being swept one after another
+      // requires a ton of boilerplate code that is hard to follow and update.
+      // Testing multiple-deposits scenarios with mocked bridge is way easier.
+      // This function prepares a fixture separate from the main test setup's
+      // fixture, just for testing multiple-deposits scenarios.
+      const prepareFixture = async function (): Promise<Fixture> {
+        const mockBank = await smock.fake<Bank>("Bank")
+        const mockBridge = await smock.fake<Bridge>("Bridge")
+
+        const TBTCFactory = await ethers.getContractFactory("TBTC")
+        // eslint-disable-next-line @typescript-eslint/no-shadow
+        const tbtc = await TBTCFactory.connect(deployer).deploy()
+
+        const TBTCVaultFactory = await ethers.getContractFactory("TBTCVault")
+        // eslint-disable-next-line @typescript-eslint/no-shadow
+        const tbtcVault = await TBTCVaultFactory.connect(deployer).deploy(
+          mockBank.address,
+          tbtc.address,
+          mockBridge.address
+        )
+
+        await mockBank.connect(deployer).updateBridge(mockBridge.address)
+        await tbtc.connect(deployer).transferOwnership(tbtcVault.address)
+        await tbtcVault.connect(deployer).addMinter(minter.address)
+
+        // Fund the `mockBank` account so it's possible to mock sending requests
+        // from it.
+        await deployer.sendTransaction({
+          to: mockBank.address,
+          value: ethers.utils.parseEther("100"),
+        })
+
+        return {
+          mockBank,
+          mockBridge,
+          tbtc,
+          tbtcVault,
+        }
+      }
+
+      context("when both deposits were optimistically minted", () => {
+        let f: Fixture
```

Comment:
Maybe just `fixture`? The `f` says nothing.

Reply:
I started with `fixture` but then all the function calls in `before` and in assertions got quite long and IMO harder to follow. My reasoning was that the `f: Fixture` is so close to the code using `f` that it's easier to find what `f` is.